R/author.R

```diff
@@ -0,0 +1,138 @@
+#' Add an author to the `Authors@R` field in DESCRIPTION
+#'
+#' @description
+
+#' `use_author()` adds a person to the `Authors@R` field of the DESCRIPTION
+#' file, creating that field if necessary. It will not modify, e.g., the role(s)
+#' or email of an existing author (judged using their "Given Family" name). For
+#' that we recommend editing DESCRIPTION directly. Or, for programmatic use,
+#' consider calling the more specialized functions available in the \pkg{desc}
+#' package directly.
+#'
+#' `use_author()` also surfaces two other situations you might want to address:
+
+#' * Explicit use of the fields `Author` or `Maintainer`. We recommend switching
+#' to the more modern `Authors@R` field instead, because it offers richer
+#' metadata for various downstream uses. (Note that `Authors@R` is *eventually*
+#' processed to create `Author` and `Maintainer` fields, but only when the
+#' `tar.gz` is built from package source.)
+
+#' * Presence of the fake author placed by [create_package()] and
+#' [use_description()]. This happens when \pkg{usethis} has to create a
+#' DESCRIPTION file and the user hasn't given any author information via the
+#' `fields` argument or the global option `"usethis.description"`. The
+#' placeholder looks something like `First Last <first.last@example.com> [aut,
+#' cre] (YOUR-ORCID-ID)` and `use_author()` offers to remove it in interactive
+#' sessions.
+#'
+#' @inheritParams utils::person
+#' @export
+#' @examples
+#' \dontrun{
+#' use_author(
+#'   given = "Lucy",
+#'   family = "van Pelt",
+#'   email = "lucy@example.com",
+#'   comment = c(ORCID = "LUCY-ORCID-ID")
+#' )
+#' }
+#'
+use_author <- function(given = NULL, family = NULL, ..., role = "ctb") {
```

Comment:
I thought about it and did take a look at author info for some existing CRAN packages. It feels pretty common to only include email for `cre` and I'm not sure there's a real purpose for including `email` for other contributors? Anyhow, that's how I ended up here.

So I guess I'm assuming that `use_author()` is going to be used mostly for adding folks in a non-`cre` role, i.e. that the maintainer is probably already set up. And that it's fine to omit `email` in this case.

This did inspire me to add `#' @inheritDotParams utils::person`, which leads to much better documentation of `...`, so that is a big improvement.

Reply:
That all makes sense to me!

Reply:
I worked on the examples a bit, so use of `email` and `role` is demonstrated, but I think most calls will be more like the last one: `use_author("Charlie", "Brown")`.

Reply:
Yeah, I like that